internal/resource/rabbitmq_plugins.go

```diff
@@ -0,0 +1,97 @@
+package resource
+
+import (
+	"strings"
+
+	rabbitmqv1beta1 "github.com/rabbitmq/cluster-operator/api/v1beta1"
+	"github.com/rabbitmq/cluster-operator/internal/metadata"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var requiredPlugins = []string{
+	"rabbitmq_peer_discovery_k8s", // required for clustering
+	"rabbitmq_prometheus",         // enforce prometheus metrics
+	"rabbitmq_management",
+}
+
+const pluginsConfigMapName = "plugins-conf"
+
+type RabbitMQPlugins struct {
+	requiredPlugins   []string
+	additionalPlugins []string
+}
+
+type RabbitmqPluginsConfigMapBuilder struct {
+	Instance *rabbitmqv1beta1.RabbitmqCluster
+}
+
+func NewRabbitMQPlugins(plugins []rabbitmqv1beta1.Plugin) RabbitMQPlugins {
+	additionalPlugins := make([]string, len(plugins))
+	for i := range additionalPlugins {
+		additionalPlugins[i] = string(plugins[i])
+	}
+
+	return RabbitMQPlugins{
+		requiredPlugins:   requiredPlugins,
+		additionalPlugins: additionalPlugins,
+	}
+}
+
+func (r *RabbitMQPlugins) DesiredPlugins() []string {
+	allPlugins := append(r.requiredPlugins, r.additionalPlugins...)
+
+	check := make(map[string]bool)
```

Comment:
I see that you are deduping the list of plugins for the value for running the command. Why not do the same thing for the values in the configMap? Am I missing something? You can extract the dedup logic and use it in `pluginsValue()`

Reply:
The config map also takes in the deduped value. The asString in plugins returns deduped list as string. Maybe should name it better to make it clearer. 

I can't think of a good name for `pluginsValue`. Can you suggest something?

Reply:
Sorry I missed it! I think the name is fine. I need to be more careful when reviewing the code.